### PR TITLE
Add linked block headers plugin

### PR DESCRIPTION
**Pull request by johndgiese**
Closes https://github.com/innolitics/n2y/issues/50

### CODE REVIEW

File: tests/test_blocks.py [hunk 1]
```diff
@@ -21,16 +21,16 @@
 )
 
 
-def generate_block(notion_block):
+def generate_block(notion_block, plugins=None):
```
Comment (johndgiese) [inline tests/test_blocks.py:24]: 👍 nice

File: n2y/plugins/mermaid.py [hunk 2]
```diff
@@ -43,7 +43,8 @@ def to_pandoc(self):
         os.write(temp_config_fd, json.dumps(mermaid_config).encode("utf-8"))
         os.close(temp_config_fd)
         try:
-            diagram_as_bytes = self.rich_text.to_plain_text().encode()
```
Comment (johndgiese) [inline n2y/plugins/mermaid.py:46]: Nice cleanup!

File: README.md [hunk 3]
```diff
@@ -137,6 +137,10 @@ This plugin assumes that the `mmdc` mermaid commandline tool is available, and w
 
 If there are errors with the mermaid syntax, it is treated as a normal codeblock and the warning is logged.
 
+### Linked Header Blocks
+
+Replace headers with links back to the originating notion block.
```
Comment (johndgiese) [inline README.md:142]: This makes it sound like the headers are no longer headers as they're replaced by links. Perhaps say "Make headers have links back to the originating notion block."

File: tests/test_linkedheaders.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            )
+        ],
+    )
+
+
+def test_heading_1():
+    notion_block = mock_block("heading_1", {"rich_text": [mock_rich_text("Heading One")]})
+    pandoc_ast, markdown = process_block(notion_block, ["n2y.plugins.linkedheaders"])
+    assert pandoc_ast == mock_header_ast(1, "One", notion_block)
+    assert markdown == f'# [Heading One](#{strip_hyphens(notion_block["id"])})\n'
```
Comment (johndgiese) [inline tests/test_linkedheaders.py:32]: What's the benefit of using what you've got over this:

```
f'# [Heading One]({notion_block["notion_url"]})\n'
```

(or equivalent)?

**Review by johndgiese — COMMENTED** [review]
Hey @bimbashrestha I can't "approve" since I created the PR, but this looks good! I had a few small suggestions, but overall seems solid!

File: tests/test_linkedheaders.py (hunk 4 above)
Comment (bimbashrestha) [reply tests/test_linkedheaders.py:32]: Ah, I like that. I didn't see "url" in the mocked notion block dict but perhaps we should add it in?

Comment (johndgiese) [reply tests/test_linkedheaders.py:32]: Ahh, I thought the url was included in the data from Notion. If it is and it's missing, I think we should add it. If it's not included in the payload from Notion, then we shouldn't and what you have now is great. I definitely want the mocks to match the notion api as best as possible.

**Pull request merged**